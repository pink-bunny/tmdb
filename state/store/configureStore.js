import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
