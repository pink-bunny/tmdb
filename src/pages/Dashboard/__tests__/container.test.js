import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import {
  fetchTrendingMovies,
  searchMovies
} from '../../../../state/dashboard/actions';
import ConnectedDashboard from '../container';

jest.mock('../../../../state/dashboard/selectors', () => ({
  isDashboardMoviesLoading: jest.fn(() => false),
  dashboardMoviesList: jest.fn(() => [{
    id: 1,
    title: 'Title',
    overview: 'Overview',
    poster_path: 'Path' /* eslint-disable-line camelcase */
  }]),
  dashboardMoviesError: jest.fn(() => ''),
  dashboardMoviesTotalPages: jest.fn(() => 21),
  dashboardMoviesCurrentPage: jest.fn(() => 1),
  dashboardMoviesSearchQuery: jest.fn(() => 'query')
}));

jest.mock('../../../../state/dashboard/actions', () => ({
  fetchTrendingMovies: jest.fn(),
  searchMovies: jest.fn()
}));

describe('Dashboard container', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();
  const wrapper = shallow(<ConnectedDashboard store={store} />).dive().dive();
  const instance = wrapper.instance();

  it('matches snapsot', () => {
    expect(instance).toMatchSnapshot();
  });

  it('componentDidMount()', () => {
    instance.componentDidMount();
    expect(fetchTrendingMovies).toHaveBeenCalled();
  });

  describe('checkSearchState method', () => {
    it('called with search query and page=2', () => {
      const { searchQuery } = instance.props;
      const page = 2;
      instance.checkSearchState(page);
      expect(searchMovies).toHaveBeenCalledWith(searchQuery, page);
    });

    it('has empty search query and page=3', () => {
      const page = 3;

      wrapper.setProps({ searchQuery: '' });
      instance.checkSearchState(page);
      expect(fetchTrendingMovies).toHaveBeenCalledWith(page);
    });
  });
});
