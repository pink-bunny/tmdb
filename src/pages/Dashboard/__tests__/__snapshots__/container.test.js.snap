// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard container matches snapsot 1`] = `
DashboardPage {
  "checkSearchState": [Function],
  "context": Object {},
  "props": Object {
    "currentPage": 1,
    "error": "",
    "fetchMovies": [Function],
    "list": Array [
      Object {
        "id": 1,
        "overview": "Overview",
        "poster_path": "Path",
        "title": "Title",
      },
    ],
    "loading": false,
    "searchMovies": [Function],
    "searchQuery": "query",
    "store": Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    },
    "totalItems": 21,
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <DashboardPage
        currentPage={1}
        error=""
        fetchMovies={[Function]}
        list={
          Array [
            Object {
              "id": 1,
              "overview": "Overview",
              "poster_path": "Path",
              "title": "Title",
            },
          ]
        }
        loading={false}
        searchMovies={[Function]}
        searchQuery="query"
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [MockFunction] {
              "calls": Array [
                Array [
                  undefined,
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        totalItems={21}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <Dashboard
        currentPage={1}
        error=""
        fetchMovies={[Function]}
        list={
          Array [
            Object {
              "id": 1,
              "overview": "Overview",
              "poster_path": "Path",
              "title": "Title",
            },
          ]
        }
        loading={false}
        totalItems={21}
      />,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
